File: get_correct_answer.py
# encoding: utf-8

import requests
import bs4 
import time
import json

#读取文件
def read_file(_file):
    with open(_file, 'r', encoding='utf-8') as infile:
        return json.load(infile)

#保存文件
def save_file(_data,_file):
    with open(_file, 'w', encoding='utf-8') as outfile:
        outfile.write(_data)


input_url = input("输入url地址:")
url = (input_url + "&pageSize=50").replace('startAnswerQuestion','getQuestions')

infile = read_file('data.json') #从data.json文件中读取题库

if infile:
    response = requests.get(url)

    questions = ''
    if response.status_code == 200:
        soup = bs4.BeautifulSoup(response.text,"html.parser") #使用pyhton内置库解析
        for content in soup.find_all('div',attrs={'class':'txtMg'}):
            question_content = content.find('span').get_text() #找到题目内容
            question_id = content.find('input',attrs={'type':'radio'}).get('class')[0][9:]  #找到题目编号

            question = "="*50 + "\n" + question_content + "\n\n" + infile[question_id]['answer'] + "\n"  # 一条问题和答案
            questions += question #添加到所有问题

    save_file(questions,'get_correct_answer.txt')  #匹配到的题目与答案保存到文件answer.txt

    print("生成答案成功! 答案已保存到：'get_correct_answer.txt'")
else:
    print("题库错误!")

File: get_correct_answer.go
package main

import (
	"encoding/json"
	"flag"
	"fmt"
	"io/ioutil"
	"log"
	"os"
	"strings"

	"github.com/anaskhan96/soup"
)

var infile = flag.String("i", "data.json", "题库文件")
var outfile = flag.String("o", "get_correct_answer.txt", "答案文件")
var usage = `
Usage: gml2json [options...]

Options:
  -i    题库文件. (default: "data.json")
  -o    答案文件. (default: "get_correct_answer.txt")

Example:

  get_correct_answer -i data.json -o get_correct_answer.txt
`

func init() {
	flag.Usage = func() {
		fmt.Fprint(os.Stderr, usage)
	}
	flag.Parse()
}

func main() {
	fmt.Printf("输入url地址:")
	var url string
	fmt.Scanf("%s", &url)
	if url != "" {
		url += "&pageSize=50"
		url = strings.Replace(url, "startAnswerQuestion", "getQuestions", -1)

		jsonstr := readfile(infile)

		str := map[string]map[string]string{}
		json.Unmarshal(jsonstr, &str)

		//log.Println(str[1])
		resp, err := soup.Get(url)
		if err != nil {
			log.Fatalf("URL读取失败.%v", err)
		}
		var questions string
		doc := soup.HTMLParse(resp)
		for _, content := range doc.FindAll("div", "class", "txtMg") {
			question_content := content.Find("span").Text()                            //找到题目内容
			question_id := content.Find("input", "type", "radio").Attrs()["class"][9:] //找到题目编号

			question := strings.Repeat("=", 50) + "\r\n"
			question += question_content + "\r\n\r\n"
			question += str[question_id]["answer"] + "\r\n" //匹配题库答案
			questions += question
		}
		savefile(*outfile, questions)
	}
}

//读取题库文件
func readfile(filename *string) []byte {
	ifile, err := ioutil.ReadFile(*filename)
	if err != nil {
		log.Fatalf("读取题库文件: [%s] 出错! %v\n", *filename, err)
	}
	log.Printf("已成功读取题库文件: [%s] ", *filename)

	return []byte(ifile)
}

//保存答案文件
func savefile(file, context string) {
	ofile, err := os.OpenFile(file, os.O_WRONLY|os.O_TRUNC|os.O_CREATE, 0666)
	if err != nil {
		log.Fatalln(err)
	}
	defer ofile.Close()

	ofile.WriteString(context)
	log.Printf("成功保存答案文件: [%s] \n", file)
}
